pkg/controllers/management/node/cleanup.go

```diff
@@ -201,7 +201,15 @@ func (m *Lifecycle) waitUntilJobCompletes(userContext *config.UserContext, job *
 	}
 
 	// remove the job to clean up
-	return userContext.K8sClient.BatchV1().Jobs(job.Namespace).Delete(m.ctx, job.Name, metav1.DeleteOptions{})
+	logrus.Infof("[node-cleanup] Cleaning job %s and related pods", job.Name)
+	if err := userContext.K8sClient.BatchV1().Jobs(job.Namespace).Delete(m.ctx, job.Name, metav1.DeleteOptions{}); err == nil {
+		logrus.Debugf("[node-cleanup] Deleting pods for job %s", job.Name)
+		return userContext.K8sClient.CoreV1().Pods(job.Namespace).DeleteCollection(m.ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("job-name=%s", job.Name),
+		})
+	} else {
+		return err
+	}
```

Comment:
It turns out that all you have to do is tell Kubernetes that you want to use `ForegroundDeletion`. Then the pod will get cleaned up before the job does (and the problem with the owner reference disappearing isn't a problem). It a little complicated because of pointers, but here it is.
```suggestion
	return userContext.K8sClient.BatchV1().Jobs(job.Namespace).Delete(m.ctx, job.Name, metav1.DeleteOptions{PropagationPolicy: &[]metav1.DeletionPropagation{metav1.DeletePropagationForeground}[0]})
```

Reply:
Much cleaner @thedadams. Thanks!